File: managers/src/main/java/com/github/managers/controllers/impl/SpecificationController.java
package com.github.managers.controllers.impl;

import com.github.managers.controllers.ISpecificationController;
import com.github.managers.dto.SpecificationDto;
import com.github.managers.exceptions.NotFound;
import com.github.managers.services.ISpecificationService;
import com.github.managers.utils.Logging;
import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RestController
@RequestMapping(path = "/v1/specification")
@RequiredArgsConstructor
public class SpecificationController implements ISpecificationController {

    private final ISpecificationService specificationService;

    @Override
    @HystrixCommand
    @Logging(isTime = true, isReturn = false)
    public SpecificationDto
    saveSpecification(Long productId, @Valid SpecificationDto payload) {
        return this.specificationService.create(productId, payload)
                .orElseThrow(NotFound::new);
    }

    @Override
    @HystrixCommand
    @Logging(isTime = true, isReturn = false)
    public SpecificationDto findById(Long id) {
        return this.specificationService.readById(id)
                .orElseThrow(NotFound::new);
    }

    @Override
    @HystrixCommand
    @Logging(isTime = true, isReturn = false)
    public void updateSpecification(SpecificationDto payload) {
        this.specificationService.update(payload);
    }

}

File: server-mocks/src/main/kotlin/com/github/server/mocks/controllers/SpecificationController.kt
package com.github.server.mocks.controllers

import com.github.server.mocks.payload.Specification
import com.github.server.mocks.services.ISpecificationService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/v1/specification")
class SpecificationController(private val specificationService: ISpecificationService) {

    @PostMapping(
            "/edit"
    )
    fun create(@RequestBody payload: Specification): Specification {
        return this.specificationService.create(payload)
    }

    @GetMapping(
            "/fetch/{id}"
    )
    fun readById(@PathVariable id: Long): Specification {
        return this.specificationService.readById(id)
    }

    @PutMapping(
            "/edit"
    )
    fun update(@RequestBody specification: Specification) {
        this.specificationService.update(specification)
    }

}